gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/CountryBalanceComputation.java

```diff
@@ -38,15 +41,32 @@ private CountryBalanceComputation() {
     public static Map<String, Double> computeSweCountriesBalances(Network network) {
         LOGGER.info("Computing initial SWE countries balance");
         Map<String, Double> countriesBalances = new HashMap<>();
-        runLoadFlow(network, network.getVariantManager().getWorkingVariantId());
-        Map<String, Double> bordersExchanges = computeSweBordersExchanges(network);
-        countriesBalances.put(toEic("PT"),  -bordersExchanges.get("ES_PT"));
-        countriesBalances.put(toEic("ES"), bordersExchanges.values().stream().reduce(0., Double::sum));
-        countriesBalances.put(toEic("FR"), -bordersExchanges.get("ES_FR"));
+
+        if (runLoadFlow(network, network.getVariantManager().getWorkingVariantId())) {
+            Map<String, Double> bordersExchanges = computeSweBordersExchanges(network);
+            countriesBalances.put(toEic("PT"), -bordersExchanges.get("ES_PT"));
+            countriesBalances.put(toEic("ES"), bordersExchanges.values().stream().reduce(0., Double::sum));
+            countriesBalances.put(toEic("FR"), -bordersExchanges.get("ES_FR"));
+            return countriesBalances;
+        }
+
+        // If the loadflow diverges, we use the netinterchange values instead
+        countriesBalances.put(toEic("PT"), getNetInterchange(network, CONTROL_AREA_PT));
+        countriesBalances.put(toEic("ES"), getNetInterchange(network, CONTROL_AREA_ES));
+        countriesBalances.put(toEic("FR"), getNetInterchange(network, CONTROL_AREA_FR));
 
         return countriesBalances;
     }
 
+    private static double getNetInterchange(Network network, String id) {
+        return network.getExtension(CgmesControlAreas.class).getCgmesControlArea(id).getNetInterchange();
```

Comment:
Instead of having constants for control areas id , we can filter the controlAreas witch have cgmesControlArea.getEnergyIdentificationCodeEIC() equals to the eic code of the country